client_options.go

```diff
@@ -3,68 +3,105 @@ package walletclient
 import (
 	"net/http"
 
-	"github.com/bitcoin-sv/spv-wallet-go-client/transports"
+	"github.com/bitcoinschema/go-bitcoin/v2"
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bk/wif"
+	"github.com/pkg/errors"
 )
 
-// WithXPriv will set xPrivString on the client
-func WithXPriv(xPrivString string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.xPrivString = xPrivString
-		}
-	}
+// WalletClientConfigurator is the interface for configuring WalletClient
+type WalletClientConfigurator interface {
+	Configure(c *WalletClient)
 }
 
-// WithXPub will set xPubString on the client
-func WithXPub(xPubString string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.xPubString = xPubString
-		}
+// XPriv sets the xPrivString field of a WalletClient
+type XPriv struct {
```

Comment:
I'm not really convinced to this struct/interface approach. IMHO it can be done with functions returning functions and would be more concise. But it's a matter of taste, so I can leave with that if no one agree with me.